hydromt_wflow/wflow.py

```diff
@@ -3274,17 +3336,47 @@ def write_forcing(
             # file_path=r'c:\Users\eilan_dk\work\profile2.html')
 
     def read_states(self):
-        """Read states at <root/?/> and parse to dict of xr.DataArray."""
+        """Read states at <root/instate/> and parse to dict of xr.DataArray."""
+        fn_default = join(self.root, "instate", "instates.nc")
+        fn = self.get_config("state.path_input", abs_path=True, fallback=fn_default)
+
         if not self._write:
             # start fresh in read-only mode
             self._states = dict()
-        # raise NotImplementedError()
 
-    def write_states(self):
-        """Write states at <root/?/> in model ready format."""
+        if fn is not None and isfile(fn):
+            self.logger.info(f"Read states from {fn}")
+            ds = xr.open_dataset(fn, mask_and_scale=False)
+            for v in ds.data_vars:
+                self.set_states(ds[v])
+
+    def write_states(self, fn_out: Union[str, Path] = None):
+        """Write states at <root/instate/> in model ready format."""
         if not self._write:
             raise IOError("Model opened in read-only mode")
-        # raise NotImplementedError()
+
+        if self.states:
+            self.logger.info("Writting states file")
+
+            # get output filename
+            if fn_out is not None:
+                self.set_config("state.path_input", fn_out)
+                self.write_config()  # re-write config
+            else:
+                fn_out = self.get_config("state.path_input", abs_path=True)
```

Comment:
A fall back is necessary here I think, something this maybe?:
```python
                fn_out = self.get_config(
                    "state.path_input", 
                    fallback=Path(
                        self.root,
                        "instate",
                        "instates.nc",
                    ),
                    abs_path=True,
                )
```

Reply:
Good catch! Solved it now.